polaris-core/src/main/java/org/apache/polaris/core/PolarisConfiguration.java

```diff
@@ -101,25 +100,6 @@ public PolarisConfiguration<T> build() {
     }
   }
 
-  /**
-   * Returns the value of a `PolarisConfiguration`, or the default if it cannot be loaded. This
-   * method does not need to be used when a `CallContext` is already available
-   */
-  public static <T> T loadConfig(PolarisConfiguration<T> configuration) {
```

Comment:
Why was this removed?

Reply:
Because 1) it relies on thread locals and 2) it can be easily inlined: it starts from a `CallContext`, then navigates the object graph until it gets to a `PolarisConfigurationStore` and calls `getConfiguration()`.

The whole method is thus generally replaceable by:

```java
@Inject PolarisConfigurationStore configurationStore;

public T loadConfig(@Nullable RealmId realmId, PolarisConfiguration<T> configuration) {
    return configurationStore.getConfiguration(realmId, configuration);
}
```

Reply:
It doesn't necessarily have to rely on thread-local, couldn't the callcontext just have been injected in?

The caller won't always have a `RealmId` available

Reply:
Indeed, the `realmId` parameter is nullable since not every call site has access to that. Without it, we always fetch the default config without considering any realm overrides.

But anyways this is going to be reverted, so I don't see the point of keeping the discussion going on.